akka-cluster-sharding-typed/src/test/scala/docs/delivery/ShardingDocExample.scala

```diff
@@ -0,0 +1,223 @@
+import akka.actor.typed.delivery.ConsumerController
+import akka.actor.typed.scaladsl.ActorContext
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.typed.delivery.ShardingConsumerController
+import akka.util.Timeout
+
+//#imports
+
+object ShardingDocExample {
+
+  //#consumer
+  trait DB {
+    def save(id: String, value: TodoList.State): Future[Done]
+    def load(id: String): Future[TodoList.State]
+  }
+
+  object TodoList {
+    import ConsumerController.SeqNr
+
+    sealed trait Command
+
+    final case class AddTask(item: String) extends Command
+    final case class CompleteTask(item: String) extends Command
+
+    private final case class InitialState(state: State) extends Command
+    private final case class SaveSuccess(seqNr: SeqNr, confirmTo: ActorRef[ConsumerController.Confirmed])
+        extends Command
+    private final case class DBError(cause: Throwable) extends Command
+
+    private final case class CommandDelivery(
+        command: Command,
+        seqNr: SeqNr,
+        confirmTo: ActorRef[ConsumerController.Confirmed])
+        extends Command
+
+    final case class State(tasks: Vector[String])
+
+    def apply(
+        id: String,
+        db: DB,
+        shardingConsumerController: ActorRef[ConsumerController.Start[Command]]): Behavior[Command] = {
+      Behaviors.setup[Command] { context =>
+        new TodoList(context, id, db).start(shardingConsumerController)
+      }
+    }
+
+  }
+
+  class TodoList(context: ActorContext[TodoList.Command], id: String, db: DB) {
+    import ConsumerController.SeqNr
+    import TodoList._
+
+    private def start(shardingConsumerController: ActorRef[ConsumerController.Start[Command]]): Behavior[Command] = {
+      context.pipeToSelf(db.load(id)) {
+        case Success(value) => InitialState(value)
+        case Failure(cause) => DBError(cause)
+      }
+
+      Behaviors.receiveMessagePartial {
+        case InitialState(state) =>
+          val deliveryAdapter: ActorRef[ConsumerController.Delivery[Command]] = context.messageAdapter { delivery =>
+            CommandDelivery(delivery.msg, delivery.seqNr, delivery.confirmTo)
+          }
+          shardingConsumerController ! ConsumerController.Start(deliveryAdapter)
+          active(state)
+        case DBError(cause) =>
+          throw cause
+      }
+    }
+
+    private def active(state: State): Behavior[Command] = {
+      Behaviors.receiveMessagePartial {
+        case CommandDelivery(AddTask(item), seqNr, confirmTo) =>
+          val newState = state.copy(tasks = state.tasks :+ item)
+          save(newState, seqNr, confirmTo)
+          active(newState)
+        case CommandDelivery(CompleteTask(item), seqNr, confirmTo) =>
+          val newState = state.copy(tasks = state.tasks.filterNot(_ == item))
+          save(newState, seqNr, confirmTo)
+          active(newState)
+        case SaveSuccess(seqNr, confirmTo) =>
+          confirmTo ! ConsumerController.Confirmed(seqNr)
```

Comment:
maybe it would be much more convenient to just have a `ActorRef[Done]` for the confirmTo? Just one thing to keep around.

Reply:
Hmm, yeah because it is anyway only one in flight, so the `seqNr` isn't that interesting?

Reply:
Ah, you already did change this...